Sources/Phoenix/Channel.swift

```diff
@@ -556,6 +556,7 @@ extension Channel {
                 }
                 
                 self.state = .joined(joinRef)
+                self.joinedOnce = true
```

Comment:
@atdrendel I had left this out as I don't think it matters much for us and because I wanted a second call to `join` to be a no-op instead of raising an exception. Why did you think it was necessary to include in our library?

Reply:
I've removed it, but I was toying around with adding completions directly to `joinPush` and `leavePush` to make our implementation mirror the JavaScript one. In that case, we would have wanted to revert to their "you can only join once" behavior. After playing around with it, I decided against doing it because it doesn't make much sense with our Combine-focused API.

https://github.com/shareup/phoenix-apple/pull/4/commits/e70eece2d4db8d66af74364e9ee9199935376ffe